### DISCUSSION TITLE
Why does GraphRAG need an embedding model, and at which stages are embeddings and LLMs invoked?

### DISCUSSION THREAD

**Discussion by kShines** [microsoft/graphrag#561] [Q&A]
Why does GraphRAG need an embedding model, and at which stages are embeddings and LLMs invoked?

**Comment by natoverse** [comment]
All text fields within the GraphRAG index can be embedded. By default, we embed only the entity description field, because this is used during the "local search" query process to find starting entry points to the graph for traversal. If you only intend to do "global search", these embeddings are not required. The env vars to adjust what gets embedded are `GRAPHRAG_EMBEDDING_TARGET` and `GRAPHRAG_EMBEDDING_SKIP`
